### DISCUSSION TITLE
Adding support for new models.

### DISCUSSION THREAD

**Discussion by KellanM** [KellanM/OpenAI-Api-Unreal#64] [Ideas]
I'll be adding support for the new models announced today:

- [ ] GPT-4 Turbo
- [ ] Dalle 3
- [ ] TTS

The Video tutorial is unforgivably out-of-date. I'll be updating this and the ReadMe right after.

**Comment by shawn8989** [comment]
Thank you!

**Comment by cocobeach** [comment]
That's great glad you keep updating it, it has a lot of potential for selling on the market place, I can't help thinking that using Assistants and tying your plugin to meta-humans could give NPCs a personality and a contextual purpose as well as a personality.
I saw an option for assistant in Make ChatLog, but we should be able to add an assistant ID, so we could add instructions and context, files with gameplay etc..?
Ideally part of the prompt should give the AI a systematic real time game log, and/or a screenshot of the situations, geolocation etc... that could make an efficient, NPC, maybe even a partner to the player?
Just brainstorming here, I love this initiative, please let me know if I can help, more than happy.
We could literally antagonise Meta on a shoestring if we make it happen!

**Comment by brybalicious** [comment]
@KellanM this issue has suddenly become very important, as OpenAI deprecated many of the models that this plugin uses, and the plugin is for all intents and purposes broken. Could you please update as per https://github.com/KellanM/OpenAI-Api-Unreal/issues/66

Is there a discord for faster communication on this?

Thank you!

**Reply by KellanM** [reply]
Absolutely! Apologies for the wait. I'll fix the deprecated functions asap. 


I've also added support for Dalle-3, Gpt-4-turbo, and their new TTS models, but it makes use of a separate plugin for importing the generated audio at runtime. I was holding off on pushing an update until I found a better solution, but fixing these deprecations takes precedence. Hang tight! Update in-bound.😄

**Reply by brybalicious** [reply]
Thanks @KellanM . Now I know you see the Discussions much faster than the Issues, much relief! :sweat_smile:

Thanks for your attention to this issue!

**Reply by brybalicious** [reply]
@KellanM any indication on an ETA? Thanks!

**Reply by KellanM** [reply]
hoping to push an update this weekend

**Reply by brybalicious** [reply]
awesome! I'll check it out and see if it works. Good luck!

**Reply by magiclamb** [reply]
Hi @KellanM I hope you've been doing well! 
How have things been travelling with the update? We appreciate you doing so, its certainly been an eventful time for OpenAI lately :)

**Reply by brybalicious** [reply]
@KellanM saw you pushed a commit last week. Cheers! Does this fix the deprecated models and replace them with working endpoints?

**Reply by KellanM** [reply]
@magiclamb @brybalicious 

I deprecated the blueprint function that allow you to call the completions endpoint.

In [OpenAI's API deprecation announcement](https://openai.com/blog/gpt-4-api-general-availability) and related documentation, they recommend using the chat endpoint as a replacement for all completion tasks.
![image](https://github.com/KellanM/OpenAI-Api-Unreal/assets/20115001/4dbcd852-ba4e-4696-9420-fdcb33fe610e)

You can test it out for yourself by by using the blueprint node `OpenAICallChat`

![image](https://github.com/KellanM/OpenAI-Api-Unreal/assets/20115001/c2f4ec3c-9b49-45fe-8fdc-5803c5bef543)

let me know what you think and don't hesitate to reach out if you have any questions! :)

**Reply by brybalicious** [reply]
Thanks @KellanM . I'll give it a go and let you know.  My use case was to ask for a response with a specific syntax so that I could parse the response and use it as a sort of general purpose general knowledge api. I guess I need to figure out how to provide instructions to the chat endpoint?

**Reply by brybalicious** [reply]
Hi @KellanM I tried to use it, but ran into build errors when trying to build the plugin for 5.3.2

Have you successfully rebuilt the plugin? Perhaps you could consider uploading it as a release for 5.3?

https://github.com/KellanM/OpenAI-Api-Unreal/issues/66#issuecomment-2022770537

**Reply by brybalicious** [reply]
Bump

**Reply by brybalicious** [reply]
Okay, it compiled for 5.3 via .sln for the project in Visual Studio. Seems like compiling plugins externally using UAT is broken with 5.3 - https://forums.unrealengine.com/t/ubt-bug-no-longer-able-to-compile-plugins-in-standalone-since-5-3/1318659
